spec/intro.dd

```diff
@@ -67,6 +67,43 @@ $(OL
 )
 
 
+$(H2 Memory Model)
+
+    $(P A $(I byte) is the fundamental unit of storage. Each byte has 8 bits. Each byte has
+    a unique address. A $(I memory location) is a sequence of one or more bytes exactly large
+    enough to hold a scalar type. Multiple threads can access separate memory locations
+    without interference.
+    )
+
+
+$(H2 Object Model)
+
+    $(P An $(I object) is created when one of the following is executed:
+    )
+
+    $(UL
+    $(LI a definition)
+    $(LI a $(GLINK2 expression, NewExpression))
+    $(LI a temporary is created)
+    $(LI changing which field of a union is active)
+    )
+
+    $(P An object spans a sequential sequence of memory locations which may or may not
+    be contiguous. It exists during construction, its lifetime, and destruction.
+    Each object has a type, which is determined either statically or by runtime
+    type information.
+    )
+
+    $(P Objects can be composed into a $(I composed object). The objects that make up
+    the composed object are $(I subobjects). An object that is not the subobject
+    of another object is a $(I complete object). The lifetime of a subobject
+    is always within the lifetime of its complete object.
+    )
+
+    $(P An object's address is the address of the first byte of the first memory
+    location for that object. Object addresses are distinct from each other
```

Comment:
How does this work for non-contiguous objects? Are non-contiguous objects always composed?

Reply:
Non-contiguous just means there may be alignment holes.

Reply:
Might want to specify that then

Reply:
The alignment issue will be addressed later.